src/Engine/ParticleEffect/ParticleEffect.cpp

```diff
@@ -0,0 +1,47 @@
+#include "ParticleEffect.hpp"
+#include <GLFW/glfw3.h>
+
+ParticleEffect::ParticleEffect(int type, int total, double duration, glm::vec3 origin) :
+    type(type),
+    total(total),
+    duration(duration),
+    origin(origin),
+    life(0),
+    toDie(false),
+    particles(generateParticles()),
+    tData(0.f),
+    t(glfwGetTime())
```

Comment:
EngineApp keeps track of run time through GLFW. I'd rather us create some system for components to access that time rather than using glfw directly

Reply:
Changed ParticleEffect to use external dt.